core/src/physics.rs

```diff
@@ -238,17 +253,24 @@ fn update_kinematic_bodies(
         {
             let mut transform = transforms.get(entity).copied().unwrap();
 
-            // Don't get stuck floating in fall-through platforms
-            if body.velocity == Vec2::ZERO
-                && collision_world.tile_collision_filtered(transform, body.shape, |ent| {
-                    collision_world
-                        .tile_collision_kinds
-                        .get(ent)
-                        .map(|x| *x == TileCollisionKind::JumpThrough)
-                        .unwrap_or(false)
-                }) == TileCollisionKind::JumpThrough
-            {
-                body.fall_through = true;
+            // If not moving, this collision test should give the same result, and will not change the value of fall_through.
+            // for players, fall_through may be modified based on inputs, and we may actually want this to be set again here,
+            // so only skip if not moving for bodies that are not players.
+            let is_player = player_states.get(entity).is_some();
+            if has_moved || is_player {
```

Comment:
This `is_player` check is something I could see being a footgun for other player-controlled items that may be created later.

I'm open to other ideas, but what if we added a flag to the kinematic body like `is_controlled`, with a note that indicates how it effects sleeping.

Then we just spawn players with their kinematics bodies `is_controlled: true`, and any other player-controlled items can use that too.

Reply:
This all sounds good to me.